pkgs/pkgs-lib/formats.nix

```diff
@@ -48,13 +48,19 @@ rec {
 
   };
 
-  # YAML has been a strict superset of JSON since 1.2
   yaml = {}:
     let jsonSet = json {};
     in jsonSet // {
       type = jsonSet.type // {
         description = "YAML value";
       };
+      generate = name: value: pkgs.runCommandNoCC name {
+        nativeBuildInputs = [ pkgs.yq ];
+        value = builtins.toJSON value;
+        passAsFile = [ "value" ];
+      } ''
+        yq --yaml-output '.' "$valuePath" > $out
```

Comment:
The toml generator already uses remarshal so I would suggest using that instead of yq:

```suggestion
        nativeBuildInputs = [ pkgs.remarshal ];
        value = builtins.toJSON value;
        passAsFile = [ "value" ];
      } ''
        json2yaml "$valuePath" "$out"
```

I think this is pretty important since remarshal and yq both have pretty big closures:

```console
$ nix path-info -Sh nixpkgs.yq nixpkgs.jq nixpkgs.remarshal
/nix/store/a209ks8rkvi2hjj895w951s2q6ivzs9b-remarshal-0.14.0   	 105.7M
/nix/store/fxdf9kizs8bwvkibr3x3zyw5yvmw80ji-python3.8-yq-2.12.0	 120.7M
/nix/store/nzsa51ydpiswbff70bczkb3hp87w0sjm-jq-1.6-bin         	  33.8M
```

At some point I started working on a small Rust application that performs something similar to remarshal but has a smaller closure (and only supports json, toml, and yaml):

```console
$ nix path-info -Sh nixpkgs.kvconv
/nix/store/3l306lhsz7vv0j46kiywj8zp5kny3cpb-kvconv-0.1.0	  32.2M
```

It also runs quite a lot faster than remarshal (mainly due to the time needed to start up Python), which would help when building many configuration files. It's not quite complete though, but if there is interest I can put a bit of effort to get it in a publishable state.

Reply:
The closures overlap though. I get

146499536 bytes for yq + remarshal
129094960 bytes for remarshal, so
17404576 bytes extra for adding yq or
~17 MB

That doesn't seem as bad.
I don't have a kvconv in my nixpkgs, so I can't determine the overlap.

`nix path-info` doesn't give you a total so I've used my `closure-size` script. You can feed it store paths as arguments.

```
#!/usr/bin/env bash
nix-store -qR "$@" \
  | ( t=0;
      while read D; do
         s=$(nix-store -q --size $D);
         printf "%10s %s\n" $s $D; t=$[t+s];
      done;
      printf "%10s %s\n" $t total;
    ) \
  | sort -n
```

Reply:
I've tested `json2yaml`, that's pretty nice and straightforward. Changed that part too.